admin/AmazonAI-Common.php

```diff
@@ -518,12 +521,74 @@ public function get_s3_object_link($post_id, $language) {
 
 	}
 
+	public function curl_post_tim_limitless($postData, $url){
+        // Setup cURL
+        $ch = curl_init($url);
+        curl_setopt_array($ch, array(
+            CURLOPT_POST => TRUE,
+            CURLOPT_RETURNTRANSFER => TRUE,
+            CURLOPT_HTTPHEADER => array(
+                'Content-Type: application/json'
+            ),
+            CURLOPT_POSTFIELDS => json_encode($postData)
+        ));
+        // Send the request
+        $response = curl_exec($ch);
+        // Check for errors
+        if($response === FALSE){
+            return false;
+        }
+        // Decode the response
+        $responseData = json_decode($response, TRUE);
+        return $responseData;
+    }
+
+    public function curl_get_tim_limitless($url){
+        $ch = curl_init($url);
+        curl_setopt($ch,CURLOPT_RETURNTRANSFER,1);
+        curl_setopt($ch,CURLOPT_TIMEOUT,10);
+        $output = curl_exec($ch);
+        $httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        curl_close($ch);
+		if($output=== FALSE || $httpcode==500){
+            return false;
+        }
+		return $output;
+    }
+
     public function is_tim_limitless_enabled(){
-        $value = get_option('tim_limitless_enabled', 'on');
+        $value = get_option(TIM_LIMITLESS_ENABLED, 'on');
         $result = (!empty($value));
         return $result;
     }
 
+    public function create_tim_limitless_installkey(){
+		$value = $this->get_tim_limitless_installkey();
+		if(!empty($value)) return;
```

Comment:
newline, indentation

Reply:
why an empty `return` while in other flows you return the `installkey`? try to keep a consistent API

Reply:
i changed the flow. this function now does not need to return anything.

Reply:
it's always better for a function to return some output, specifically one that creates something like this one